hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3520,9 +3520,9 @@ def __getitem__(self, item):
             slices = []
             for i, s in enumerate(item):
                 if isinstance(s, slice):
-                    start = s.start if s.start is not None else to_expr(0, tint64)
-                    stop = s.stop if s.stop is not None else self.shape[i]
                     step = s.step if s.step is not None else to_expr(1, tint64)
+                    start = hl.cond(s.start >= 0, s.start, self.shape[i] + s.start) if s.start is not None else hl.cond(step >= 0, to_expr(0, tint64), self.shape[i] - 1)
+                    stop = hl.cond(s.stop >= 0, s.stop, self.shape[i] + s.stop) if s.stop is not None else hl.cond(step >= 0, self.shape[i], to_expr(-1, tint64))
```

Comment:
Admittedly these two lines are kind of a lot, but didn't find a clearer way to write them.

Reply:
They might be clearer if not written with the ternary operator and instead as
```python3
if ...:
  step = ..
else:
  step = ...
```
But I don't have strong feelings here.